tests/map/test-map.c

```diff
@@ -0,0 +1,101 @@
+#include <assert.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+
+#include "map.h"
+
+static void swap(int *x, int *y)
+{
+    int tmp = *x;
+    *x = *y;
+    *y = tmp;
+}
+
+enum { N_NODES = 10000 };
+
+bool test_mix_insert()
+{
+    bool failed = false;
```

Comment:
The variable `failed` should be updated accordingly.
When the program is compiled with `NDEBUG` set, macro `assert` would be no operation, and we do need the result of this function for validation.